OneSignalSDK/onesignal/location/src/main/java/com/onesignal/location/internal/LocationManager.kt

```diff
@@ -179,13 +179,18 @@ internal class LocationManager(
 
     // Started from this class or PermissionActivity
     private suspend fun startGetLocation() {
-        Logging.debug("LocationManager.startGetLocation()") // with lastLocation: " + lastLocation)
-        try {
-            if (!_locationController.start()) {
-                Logging.warn("LocationManager.startGetLocation: not possible, no location dependency found")
+        if (isShared) {
```

Comment:
Recommend returning early instead if `isShared` is `false`, this keeps the code from getting to nested.